lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -86,25 +108,148 @@ void CompRegOp::print(::mlir::OpAsmPrinter &p) {
+
+  Op input, clk;
+  StringAttr innerSym;
+
+  if (parser.parseOperand(input) || parser.parseComma() ||
+      parser.parseOperand(clk) || parser.parseComma() ||
+      parser.parseSymbolName(innerSym, "inner_sym", result.attributes))
+    return failure();
+
+  // Parse reset [sync|async] %reset, %value
+  Optional<std::pair<Op, Op>> resetAndValue;
+  if (succeeded(parser.parseOptionalKeyword("reset"))) {
+    bool isAsync;
+    if (succeeded(parser.parseOptionalKeyword("async")))
+      isAsync = true;
+    else if (succeeded(parser.parseOptionalKeyword("sync")))
+      isAsync = false;
+    else
+      return parser.emitError(loc, "invalid reset, expected 'sync' or 'async'");
+    result.attributes.append("isAsync", builder.getBoolAttr(isAsync));
+
+    resetAndValue = {{}, {}};
+    if (parser.parseOperand(resetAndValue->first) || parser.parseComma() ||
+        parser.parseOperand(resetAndValue->second))
+      return failure();
   } else {
-    SmallString<32> resultNameStr;
-    llvm::raw_svector_ostream tmpStream(resultNameStr);
-    p.printOperand(data(), tmpStream);
-    auto actualName = tmpStream.str().drop_front();
-    if (actualName == name())
-      elidedAttrs.push_back("name");
+    result.attributes.append("isAsync", builder.getBoolAttr(false));
   }
 
+  if (succeeded(parser.parseOptionalKeyword("randomised")))
+    result.attributes.append("isRandomised", builder.getUnitAttr());
+
+  Type ty;
+  if (parser.parseOptionalAttrDict(result.attributes) || parser.parseColon() ||
+      parser.parseType(ty))
+    return failure();
+  result.addTypes({ty});
+
+  setNameFromResult(parser, result);
+
+  Type i1 = IntegerType::get(result.getContext(), 1);
+  if (parser.resolveOperand(input, ty, result.operands) ||
+      parser.resolveOperand(clk, i1, result.operands))
+    return failure();
+
+  if (resetAndValue) {
+    if (parser.resolveOperand(resetAndValue->first, i1, result.operands) ||
+        parser.resolveOperand(resetAndValue->second, ty, result.operands))
+      return failure();
+  }
+
+  return success();
+}
+
+void FirRegOp::print(::mlir::OpAsmPrinter &p) {
+  SmallVector<StringRef> elidedAttrs = {inner_symAttrName(), isAsyncAttrName(),
+                                        isRandomisedAttrName()};
+
+  p << ' ' << input() << ", " << clk() << ", ";
+  p.printSymbolName(inner_sym());
+
+  if (hasReset()) {
+    p << " reset " << (isAsync() ? "async" : "sync") << ' ';
+    p << reset() << ", " << resetValue();
+  }
+
+  if (isRandomised())
+    p << " randomised";
+
+  if (canElideName(p, *this))
+    elidedAttrs.push_back("name");
+
   p.printOptionalAttrDict((*this)->getAttrs(), elidedAttrs);
   p << " : " << input().getType();
 }
 
 /// Suggest a name for each result value based on the saved result names
 /// attribute.
-void CompRegOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
+void FirRegOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
   // If the wire has an optional 'name' attribute, use it.
```

Comment:
nit: this says "wire" but it's really a "reg", so I would suggest fixing it since you're touching it here.